test/_stub/AbstractControllerSpec.scala

```diff
@@ -0,0 +1,8 @@
+package _stub
+
+import play.api.mvc.ControllerComponents
+import play.api.test.Helpers.stubControllerComponents
+
+abstract class AbstractControllerSpec extends AbstractSpec {
+  protected val ccMock: ControllerComponents = stubControllerComponents()
+}
```

Comment:
This is the base class for every controller's unit test classes. This class extends to AbstractSpec and has a mock of ControllerComponent.